nodeup/pkg/model/context.go

```diff
@@ -399,7 +399,11 @@ func (c *NodeupModelContext) UseBootstrapTokens() bool {
 
 // UseSecureKubelet checks if the kubelet api should be protected by a client certificate.
 func (c *NodeupModelContext) UseSecureKubelet() bool {
-	return c.NodeupConfig.KubeletConfig.AnonymousAuth != nil && !*c.NodeupConfig.KubeletConfig.AnonymousAuth
+	if c.IsKubernetesGTE("1.21") {
+		return true
```

Comment:
I don't see much logging elsewhere in this file, but should we warn users if they had enabled anonymous authentication, and we're overriding their intention here?

Reply:
Ah, I just saw #11180, which I think takes care of catching that discrepancy.